mailbox/lucene/src/main/java/org/apache/james/mailbox/lucene/search/LuceneIndexableDocument.java

```diff
@@ -0,0 +1,274 @@
+import static org.apache.james.mailbox.lucene.search.DocumentFieldConstants.SENT_DATE_FIELD_YEAR_RESOLUTION;
+import static org.apache.james.mailbox.lucene.search.DocumentFieldConstants.SENT_DATE_SORT_FIELD_MILLISECOND_RESOLUTION;
+import static org.apache.james.mailbox.lucene.search.DocumentFieldConstants.SIZE_FIELD;
+import static org.apache.james.mailbox.lucene.search.DocumentFieldConstants.SUBJECT_FIELD;
+import static org.apache.james.mailbox.lucene.search.DocumentFieldConstants.THREAD_ID_FIELD;
+import static org.apache.james.mailbox.lucene.search.DocumentFieldConstants.TO_FIELD;
+import static org.apache.james.mailbox.lucene.search.DocumentFieldConstants.UID_FIELD;
+import static org.apache.james.mailbox.lucene.search.DocumentFieldConstants.USERS;
+
+import java.util.Arrays;
+import java.util.Date;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Optional;
+
+import jakarta.mail.Flags;
+
+import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.mailbox.MessageUid;
+import org.apache.james.mailbox.extractor.TextExtractor;
+import org.apache.james.mailbox.model.MailboxId;
+import org.apache.james.mailbox.model.MessageAttachmentMetadata;
+import org.apache.james.mailbox.opensearch.json.HeaderCollection;
+import org.apache.james.mailbox.opensearch.json.MimePart;
+import org.apache.james.mailbox.opensearch.json.MimePartParser;
+import org.apache.james.mailbox.store.mail.model.MailboxMessage;
+import org.apache.james.mailbox.store.search.SearchUtil;
+import org.apache.lucene.document.DateTools;
+import org.apache.lucene.document.Document;
+import org.apache.lucene.document.Field;
+import org.apache.lucene.document.LongPoint;
+import org.apache.lucene.document.NumericDocValuesField;
+import org.apache.lucene.document.SortedSetDocValuesField;
+import org.apache.lucene.document.StoredField;
+import org.apache.lucene.document.StringField;
+import org.apache.lucene.document.TextField;
+import org.apache.lucene.util.BytesRef;
+
+import com.github.fge.lambdas.Throwing;
+import com.google.common.collect.ImmutableMap;
+
+import reactor.core.publisher.Mono;
+
+public class LuceneIndexableDocument {
+    public static final Map<Flags.Flag, String> SYSTEM_FLAG_STRING_MAP = ImmutableMap.<Flags.Flag, String>builder()
+        .put(Flags.Flag.ANSWERED, "\\ANSWERED")
+        .put(Flags.Flag.DELETED, "\\DELETED")
+        .put(Flags.Flag.DRAFT, "\\DRAFT")
+        .put(Flags.Flag.FLAGGED, "\\FLAGGED")
+        .put(Flags.Flag.RECENT, "\\RECENT")
+        .put(Flags.Flag.SEEN, "\\FLAG")
+        .build();
+
+    private final TextExtractor textExtractor;
+
+    public LuceneIndexableDocument(TextExtractor textExtractor) {
+        this.textExtractor = textExtractor;
+    }
+
+    public Mono<Document> createMessageDocument(MailboxMessage message, MailboxSession session) {
+        return Throwing.supplier(() -> new MimePartParser(textExtractor).parse(message.getFullContent())).get()
```

Comment:
If mimePartParser is needed shall we move it into `mailbox-store` rather than adding a dependency to mailbox-opensearch